app/modules/net.c

```diff
@@ -233,7 +233,7 @@ static err_t net_tcp_recv_cb(void *arg, struct tcp_pcb *tpcb, struct pbuf *p, er
     return tcp_close(tpcb);
   }
   net_recv_cb(ud, p, 0, 0);
-  tcp_recved(tpcb, p->len);
+  tcp_recved(tpcb, ud->client.hold ? 0 : TCP_WND);
```

Comment:
is it really safe to use TCP_WND here rather than p->len (or possibly an accumulation of p->len from all packets received since the hold started)